src/atoms/Tab/Tab.native.js

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { View, TouchableOpacity } from 'react-native';
+import styled, { ThemeContext } from 'styled-components/native';
+import Flex from '../Flex';
+import Text from '../Text';
+import { getColor } from '../../_helpers/theme';
+import Icon from '../Icon';
+import Space from '../Space';
+import icons from '../../icons';
+import automation from '../../_helpers/automation-attributes';
+
+const styles = {
+  label: {
+    borderBottomWidth({ isActive }) {
+      if (isActive) {
+        return '1px';
+      }
+      return '0px';
+    },
+    borderBottomColor({ isActive, theme, disabled }) {
+      if (disabled) {
+        if (isActive) {
+          return getColor(theme, 'primary.300');
+        }
+      }
+      if (isActive) {
+        return getColor(theme, 'primary.800');
+      }
+      return 'transparent';
+    },
+    margin() {
+      return [0, 0, 0, 0.5];
+    },
+  },
+};
+
+const StyledTabButton = styled(TouchableOpacity)(
+  (props) => `
+min-width: 48px;
+background-color: ${props.backgroundColor};
+border-top-left-radius: 2px;
+border-top-right-radius: 2px;
+`,
+);
+
+const Label = styled(View)`
+  padding: 8px 0px;
+  border-bottom-width: ${styles.label.borderBottomWidth};
+  border-bottom-color: ${styles.label.borderBottomColor};
+`;
+
+const getLabelColor = ({ isActive, disabled }) => {
+  let color = 'shade.960';
+
+  if (isActive) {
+    color = disabled ? 'primary.700' : 'primary.800';
+  } else {
+    color = disabled ? 'shade.940' : 'shade.960';
+  }
+  return color;
+};
+
+const Tab = ({ label, isActive, icon, onPress, disabled, testID }) => {
+  const initialLabelColor = getLabelColor({ isActive, disabled });
+
+  const [tabBackground, setTabBackground] = useState('transparent');
+  const [labelColor, setLabelColor] = useState(initialLabelColor);
+
+  const theme = useContext(ThemeContext);
+
+  const onPressIn = () => {
+    if (isActive) {
+      setLabelColor('primary.900');
+      setTabBackground(getColor(theme, 'primary.200'));
```

Comment:
Can't we set colors through the component using only it's name like 
```suggestion
      setTabBackground('primary.200');
```

so that you don't need to get ```theme``` from context. You can resolve this color in styles above

Reply:
I have to change the background color on `pressIn` events, how do I connect the styles object with the events that are fired by the component?

Reply:
You can have a ```isPressed``` state that is used to derive colors of Label & Tab